### PR TITLE
CI: More consistent minRequired determination in WaitforNPods

### PR DESCRIPTION
**Pull request by raybejjani**
We rely on WaitforNPods to detect the count of pods to wait for very
often in the code, often including no filter (thus waiting on all pods).
Since pods may be in the process of being deleted this can result in
minRequired being too high. The minRequired determination is now more
consistent with what we would accept as a running pod.

Signed-off-by: Ray Bejjani <ray@covalent.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7993)

### CODE REVIEW

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-nightly

**Review by brb — APPROVED** [review]
LGTM! This should really decrease the likelihood of the `Timeout reached: timed out waiting for pods with filter $FILTER to be ready` flakes.

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -556,8 +556,14 @@ func (kub *Kubectl) WaitforNPods(namespace string, filter string, minRequired in
 			return false
 		}
 
+		// Determine how many pods we should wait for. We reject pods that are
```
Comment (ianvernon) [inline test/helpers/kubectl.go:559]: This is outside of the scope of this PR, but we should really have a wrapper that says 'wait for deployment to be ready' and introspect the replica count of the deployment. Then we can wait for 2 pods to be ready with the labels from the deployment.

**Review by ianvernon — APPROVED** [review]

File: test/helpers/kubectl.go (hunk 1 above)
Comment (raybejjani) [reply test/helpers/kubectl.go:559]: Hehe, I considered doing that then I decided that would be too big a change. Still, that is definitely a more reliable approach. In some cases we seem to have an empty filter passed in, and I need to figure out if the intention was "all pods are stable" or "I didn't bother figuring this out and I copied the other test"

**Comment by raybejjani** [comment]
superseded by #7995 for now

**Pull request closed without merging**